### ISSUE TITLE
Problem with deconvolution layer

### ISSUE THREAD

**Issue by vdavid70619** [BVLC/caffe#4446]
Hi,
I am using FCN to fine-tune on a new task. I notice a potential problem with the deconvolution layer in the master branch. If I just use the FCN train.prototxt provided in the repository and don't set the lr_mult for deconvolution to 1, then caffe won't update the blob diff back to previous layers  (see the table attached). This causes the training process problematic. 

               -------------------- blob diff ---------------------
                     data|       0.0000e+00      0.0000e+00      0.0000e+00
                    label|       0.0000e+00      0.0000e+00      0.0000e+00
                     mask|       0.0000e+00      0.0000e+00      0.0000e+00
      mask_mask_0_split_0|       0.0000e+00      0.0000e+00      0.0000e+00
      mask_mask_0_split_1|       0.0000e+00      0.0000e+00      0.0000e+00
                  conv1_1|       0.0000e+00      0.0000e+00      0.0000e+00
                  conv1_2|       0.0000e+00      0.0000e+00      0.0000e+00
                    pool1|       0.0000e+00      0.0000e+00      0.0000e+00
                  conv2_1|       0.0000e+00      0.0000e+00      0.0000e+00
                  conv2_2|       0.0000e+00      0.0000e+00      0.0000e+00
                    pool2|       0.0000e+00      0.0000e+00      0.0000e+00
                  conv3_1|       0.0000e+00      0.0000e+00      0.0000e+00
                  conv3_2|       0.0000e+00      0.0000e+00      0.0000e+00
                  conv3_3|       0.0000e+00      0.0000e+00      0.0000e+00
                    pool3|       0.0000e+00      0.0000e+00      0.0000e+00
                  conv4_1|       0.0000e+00      0.0000e+00      0.0000e+00
                  conv4_2|       0.0000e+00      0.0000e+00      0.0000e+00
                  conv4_3|       0.0000e+00      0.0000e+00      0.0000e+00
                    pool4|       0.0000e+00      0.0000e+00      0.0000e+00
                  conv5_1|       0.0000e+00      0.0000e+00      0.0000e+00
                  conv5_2|       0.0000e+00      0.0000e+00      0.0000e+00
                  conv5_3|       0.0000e+00      0.0000e+00      0.0000e+00
                    pool5|       0.0000e+00      0.0000e+00      0.0000e+00
                      fc6|       0.0000e+00      0.0000e+00      0.0000e+00
                      fc7|       0.0000e+00      0.0000e+00      0.0000e+00
                 score_fr|       0.0000e+00      0.0000e+00      0.0000e+00
                  upscore|       -9.3750e-01     0.0000e+00      6.2500e-02
                    score|       -9.3750e-01     0.0000e+00      6.2500e-02
                     loss|       1.0000e+00      1.0000e+00      1.0000e+00

Thanks,
Xiyang

**Comment by vdavid70619** [comment]
It turns out that  the "bilinear filler" in deconvolution doesn't seem to work. Stick back to net_surgery.

**Comment by zizhaozhang** [comment]
Hi Xiyang

Can you give a specific observations? You mean the bilinear filler is not
working? I am quite curious about that. I thought it just initializes the
weights, so it is suppose not to influence the backprop... Thanks

On Thu, Jul 14, 2016 at 5:36 PM, Xiyang Dai <notifications@github.com>
wrote:

> It turns out that the "bilinear filler" in deconvolution doesn't seem to
> work. Stick back to net_surgery.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/BVLC/caffe/issues/4446#issuecomment-232799632>, or mute
> the thread
> <https://github.com/notifications/unsubscribe/AFfrUaL4HPsDFNne70nUbPBCp-yRKp2_ks5qVqvFgaJpZM4JJe0L>
> .
>



-- 
Best Regards,
Zizhao

**Comment by bdengNE** [comment]
@vdavid70619  
Hi David, I am also in trouble with this issue, have you figured out how to solve it???
Thanks,
Bowen

**Comment by shelhamer** [comment]
Please ask usage questions on the mailing list. The bilinear filler does in fact work as long as `group` is set accordingly, as highlighted in the filler documentation: http://caffe.berkeleyvision.org/doxygen/classcaffe_1_1BilinearFiller.html#details

From https://github.com/BVLC/caffe/blob/master/CONTRIBUTING.md:

> _Please do not post usage, installation, or modeling questions, or other requests for help to Issues._
Use the [caffe-users list](https://groups.google.com/forum/#!forum/caffe-users) instead. This helps developers maintain a clear, uncluttered, and efficient view of the state of Caffe.
